sum-product.tex

```diff
@@ -1,11 +1,53 @@
 \subsection{Sum Product Algorithm}
 
-For the sake of exposition, suppose we have an undirected graphical model which is a tree and that we want to compute the marginal distribution $p(x)$ for some variable $x$.  % means we have functions on every edge and on every node?  does it necessarily mean that?
-The crux of the sum-product algorithm is that conditioning on a single parent $y\in\pi(x)$ splits the graph into two conditionally independent pieces: the tree in which $y$ participates, and the tree in which $x$ participates.  To compute the marginal for $x$, the only information we actually need from $y$ is the function $y \rightarrow \sum_{z \in T_y} p(z | y)$.  To see this, use the conditional independence to write:
+Suppose we have a graphical model which is a connected undirected tree; in that case we can choose an arbitrary ordering for the nodes and write:
 $$
-p(x, ...) \sim p(y) \cdot p(T_y|y)p(T_x|y)
+p(x_1, x_2, .., x_n) = \prod_i \psi_i(x_i) \prod_{i, j \in E} \psi_{i,j}(x_i, x_j)
 $$
-and thus
+
+Suspend disbelief about probabilistic interpretations for a moment and suppose we simply want to compute the quantity
+$$
+p(x_s) := \sum_{i \neq s} p(x_1, x_2, ..,x_s,... x_n)
+$$
+where we interpret the sum as being over the \emph{state space} of the corresponding variables.  We can write
+\begin{equation*}
+\begin{aligned}
+p(x_s) &= \psi_s(x_s) \prod_{i\neq s} \psi_i(x_i) \prod_{i, j \in E} \psi_{i,j}(x_i, x_j)\\
```

Comment:
This is a single p(x_1, x_2, ..,x_s,... x_n) term, no?``

Reply:
Yup, good catch.  Fixed